### PR TITLE
Build-OpenCV bug fixes

### PR DESCRIPTION
**Pull request by rsnk96**
* Removed `undo_rename` of the renamed conda library files since QT doesn't work with it on
* Removed renaming of
    ```
        rename_so libz.so
        rename_so libfreetype.so
        rename_so libharfbuzz.so
        rename_so libtbb.so
    ```
    @rajat2004  did these libraries cause problems for you when you tried it out? It works on a fresh installation of Ubuntu Mate 18.04 with just the renaming of `libfontconfig` and `libpangoft` for me

### CODE REVIEW

**Comment by rajat2004** [comment]
The problems were seen in the OpenCV + Conda Travis build for Ubuntu 16.04 and 18.04
I think a check should be added to test the OS version and rename libraries that are required, that PR had a lot of things, so this got missed.
Not sure if the entire `undo_rename` should be removed, since things do work after undoing the rename.

**Comment by rsnk96** [comment]
> The problems were seen in the OpenCV + Conda Travis build for Ubuntu 16.04 and 18.04
I think a check should be added to test the OS version and rename libraries that are required, that PR had a lot of things, so this got missed.

Hopefully those errors aren't there now, the Travis build isn't showing anything. Do check if you feel the same.



>  Not sure if the entire undo_rename should be removed, since things do work after undoing the rename

Though `cv2.getBuildInformation()` works with `undo_rename` there, QT windows do not work in my tests. Ex: `cv2.namedWindow()` gives an error. Did you test QT windows coming up after the `undo_rename()`?

**Comment by rajat2004** [comment]
> Hopefully those errors aren't there now, the Travis build isn't showing anything. Do check if you feel the same.

The Travis build is showing the problems (Xenial) - 
```
/opt/anaconda3/lib/libtbb.so.2: undefined reference to `__cxa_init_primary_exception@CXXABI_1.3.11'
/opt/anaconda3/lib/libtbb.so.2: undefined reference to `std::__exception_ptr::exception_ptr::exception_ptr(void*)@CXXABI_1.3.11'
collect2: error: ld returned 1 exit status
modules/xobjdetect/tools/waldboost_detector/CMakeFiles/opencv_waldboost_detector.dir/build.make:93: recipe for target 'bin/opencv_waldboost_detector' failed
```

Bionic build failed due to download error

> Though `cv2.getBuildInformation()` works with `undo_rename` there, QT windows do not work in my tests. Ex: `cv2.namedWindow()` gives an error. Did you test QT windows coming up after the `undo_rename()`?

Wasn't able to, was testing in a docker container, and didn't do the X11 windows while starting the container. Was short on time, therefore looked for alternate ways and the `ldd` info seemed correct, also from info on the net
Will do tests more thoroughly afterwards, sorry for this and thanks for fixing!

File: Build-OpenCV.sh [hunk 1]
```diff
@@ -221,9 +213,6 @@ if [[ -n $(echo $PATH | grep 'conda') ]] ; then
 
         rename_so libfontconfig.so
         rename_so libpangoft2-1.0.so
-        rename_so libz.so
-        rename_so libfreetype.so
-        rename_so libharfbuzz.so
```
Comment (rajat2004) [inline Build-OpenCV.sh:226]: Each rename was added after build failed on 16.04 in Travis, so all of them will mostly be needed
Adding an OS version check for the ones specific for 16.04 would be good though

**Comment by rajat2004** [comment]
Cancelled the Travis branch build since the PR build was queued

**Comment by rajat2004** [comment]
@rsnk96 Noticed the target branch was not `master`, changed that but then the older commits which have already been merged came up, will need to drop the commits
Would do it myself but don't want to mess up your commit history as well, had some bad experiences earlier with that

File: Build-OpenCV.sh (hunk 1 above)
Comment (rajat2004) [reply Build-OpenCV.sh:226]: The others are warnings, so might be okay to remove these

**Comment by rsnk96** [comment]
No worries, do go ahead and fix it if you can!

**Comment by rajat2004** [comment]
Seems okay to me, though I haven't done any further testing
@rsnk96 ?

**Comment by rsnk96** [comment]
Looks good to me, merging!

**Pull request merged**